haechan45th/app.js

```diff
@@ -149,13 +147,79 @@ app.get('/users/:userId/posts', async (req, res) => {
       WHERE users.id = ?
       `, [userId])
 
-    return res.status(200).json({gettingUserPosts})
-  } catch(error) {
+    return res.status(200).json({ gettingUserPosts })
+  } catch (error) {
     console.log(error)
-    res.status(400).json({message: "some error"})
+    res.status(400).json({ message: "some error" })
+  }
+});
+
+app.put('/posts/:userId', async (req, res) => {
+  try {
+    const { userId } = req.params;
+    const { postId } = req.body;
+    const { title, content, imageUrl, } = req.body;
+
+    const updatePosts = await dataSource.query(
+      `UPDATE posts
+      INNER JOIN users ON posts.user_id = users.id
+      SET
+      posts.title = ?,
+      posts.content = ?,
+      posts.image_url = ?
+      WHERE users.id = ? AND posts.id = ?
+      `, [title, content, imageUrl, userId, postId]
+    )
+
+    res.status(200).json({ message: 'Post updated successfully.' });
+  } catch (error) {
+    console.log(error);
+    res.status(400).json({ message: "Error has occur in USER EDITING API" })
+  }
+});
+
+app.delete('/posts/:postsId', async (req, res) => {
+  try {
+    const { postsId } = req.params;
+
+    const deletePosts = await dataSource.query(
+      `DELETE FROM posts
+      WHERE posts.id = ${postsId}
+      `
+    )
+
+    res.status(200).json({ message: 'Post DELETED successfully.' });
+  } catch (error) {
+    console.log(error);
+    res.status(400).json({ message: "Error has occur in DELETING USER POSTS" })
   }
 });
```

Comment:
```suggestion
app.delete('/posts/:postsId', async (req, res) => {
  try {
    const { postsId } = req.params;

    const deletePosts = await dataSource.query(
      `DELETE FROM posts
      WHERE posts.id = ? AND posts.user_id = ?
      `, [ postsId, userId ]
    )

    res.status(200).json({ message: 'Post DELETED successfully.' });
  } catch (error) {
    console.log(error);
    res.status(400).json({ message: "Error has occur in DELETING USER POSTS" })
  }
});
```
WHERE clause의 condition에는 가급적 템플릿 리터럴 (`${변수}`)을 쓰지 않도록 해주세요!
추가로 userId까지 client에서 받아 condition에 AND로 추가함으로써 보다 안전하고 정확한 삭제 query를 만들어주세요.

Reply:
옷 넵! 명심하겠습니다